bin/describe/describe_external_lib_deps.ml

```diff
@@ -65,53 +73,62 @@ let is_external db name =
     | Installed_private | Public _ | Private _ -> false
     | Installed -> true)
 
-let external_lib_pps db preprocess =
+let resolve_lib db name kind =
+  let open Memo.O in
+  let+ is_external = is_external db name in
+  if is_external then External { name; kind } else Local { name; kind }
+
+let resolve_lib_pps db preprocess =
   let open Memo.O in
   let* pps =
     Resolve.Memo.read_memo
       (Dune_rules.Preprocess.Per_module.with_instrumentation preprocess
          ~instrumentation_backend:(Dune_rules.Lib.DB.instrumentation_backend db))
     >>| Dune_rules.Preprocess.Per_module.pps
   in
-  Memo.parallel_map
-    ~f:(fun (_, name) ->
-      let+ is_external = is_external db name in
-      if is_external then Some { name; kind = Kind.Required } else None)
-    pps
-  >>| List.filter_opt
+  Memo.parallel_map ~f:(fun (_, name) -> resolve_lib db name Kind.Required) pps
 
-let external_resolve db name kind =
+let resolve_lib_deps db lib_deps =
   let open Memo.O in
-  let+ is_external = is_external db name in
-  if is_external then Some { name; kind } else None
-
-let external_lib_deps db lib_deps =
-  let open Memo.O in
-  Memo.parallel_map lib_deps ~f:(fun lib ->
+  Memo.parallel_map
+    ~f:(fun lib ->
       match lib with
-      | Dune_rules.Lib_dep.Direct (_, name) | Re_export (_, name) -> (
-        let+ v = external_resolve db name Kind.Required in
-        match v with
-        | Some x -> [ x ]
-        | None -> [])
-      | Select select ->
-        Memo.parallel_map select.choices
-          ~f:(fun (choice : Dune_rules.Lib_dep.Select.Choice.t) ->
-            Memo.parallel_map
-              (Lib_name.Set.to_string_list choice.required
-              @ Lib_name.Set.to_string_list choice.forbidden)
-              ~f:(fun name ->
-                external_resolve db (Lib_name.of_string name) Kind.Optional)
-            >>| List.filter_opt)
+      | Dune_rules.Lib_dep.Direct (_, name)
```

Comment:
You should put `(lib : Dune_rules.Lib_dep.t)` in the match then you can stop prefixing the constructors.